### PR TITLE
Output porosity stats

### PR DESCRIPTION
**Pull request by vovannikov**
Basically, this prints a list of all pores to a file.

### CODE REVIEW

File: applications/sintering/include/pf-applications/sintering/driver.h [hunk 1]
```diff
@@ -2947,10 +2947,22 @@ namespace Sintering
...
-                                            output);
+          Postprocessors::output_porosity(mapping,
+                                          dof_handler,
+                                          solution,
+                                          output);
+        }
+
+      if (params.output_data.porosity_stats)
+        {
+          std::string output = params.output_data.vtk_path +
```
Comment (peterrum) [inline applications/sintering/include/pf-applications/sintering/driver.h:2958]:
```suggestion
          const std::string output = params.output_data.vtk_path +
```

File: applications/sintering/include/pf-applications/sintering/postprocessors.h [hunk 2]
```diff
@@ -1180,11 +1180,10 @@ namespace Sintering
       run_flooding(const typename DoFHandler<dim>::cell_iterator &cell,
                    const BlockVectorType &                        solution,
                    LinearAlgebra::distributed::Vector<Number> &   particle_ids,
-                   const unsigned int                             id)
+                   const unsigned int                             id,
+                   const double threshold_lower     = 0.5,
```
Comment (peterrum) [inline applications/sintering/include/pf-applications/sintering/postprocessors.h:1184]: This is a different value!?

File: applications/sintering/include/pf-applications/sintering/postprocessors.h [hunk 3]
```diff
@@ -1338,6 +1354,59 @@ namespace Sintering
...
+    {
+      const double invalid_particle_id = -1.0; // TODO
+
+      LinearAlgebra::distributed::Vector<double> particle_ids(
+        dof_handler.get_triangulation()
+          .global_active_cell_index_partitioner()
+          .lock());
+
+      // Detect pores and assign ids
+      auto [local_to_global_particle_ids, offset] = internal::detect_pores(
```
Comment (peterrum) [inline applications/sintering/include/pf-applications/sintering/postprocessors.h:1371]: The vector `particle_ids` still seems to be empty so how is it possible to run `detect_pores`?

File: applications/sintering/include/pf-applications/sintering/postprocessors.h (hunk 2 above)
Comment (vovannikov) [reply applications/sintering/include/pf-applications/sintering/postprocessors.h]: I was experimenting with different values, but 0.8 indeed looks optimal, I set it back.

File: applications/sintering/include/pf-applications/sintering/postprocessors.h (hunk 3 above)
Comment (vovannikov) [reply applications/sintering/include/pf-applications/sintering/postprocessors.h]: `particle_ids ` is filled inside `detect_pores()`. It was indeed confusing, I have refactored that to be more explicit.

**Review by peterrum — APPROVED** [review]

**Pull request merged**